vertexai/multimodal-audio/multimodalaudio.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// multimodalaudio shows an example of understanding multimodal input including audio
+package multimodalaudio
+
+// [START generativeaionvertexai_gemini_audio_summarization]
+// [START generativeaionvertexai_gemini_audio_transcription]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"path/filepath"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// [END generativeaionvertexai_gemini_audio_transcription]
+// [END generativeaionvertexai_gemini_audio_summarization]
+
+// [START generativeaionvertexai_gemini_audio_summarization]
+// summarizeAudio generates a response into w, based upon the prompt
+// and audio provided.
+// audio is a Google Cloud Storage path starting with "gs://"
+func summarizeAudio(w io.Writer, prompt, audio, projectID, location, modelName string) error {
+	// prompt := `
+	// 		Please provide a summary for the audio.
+	// 		Provide chapter titles with timestamps, be concise and short, no need to provide chapter summaries.
+	// 		Do not make up any information that is not part of the audio and do not be verbose.
+	// `
+	// audio := "gs://cloud-samples-data/generative-ai/audio/pixel.mp3"
+	// location := "us-central1"
+	// modelName := "gemini-1.5-pro-preview-0409"
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, projectID, location)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %v", err)
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel(modelName)
+	model.SetTemperature(0.4)
```

Comment:
Is there a reason for this particular value? For changing the default temperature at all?

Reply:
No good reason, thanks for finding this!

My understanding is the default temp in the client lib model is `nil`, which means the endpoint model's default will be used (it depends on the model name).  In this case, not setting the temperature is the right thing to do, and it does not mean "temp=0".

Reply:
Correct. IIRC the default is like 0.9.

Reply:
In an earlier sample, temperature was included as a parameter and I suggested hard-coding it. This doesn't need to happen everywhere, but it does surface that the toggle exists while keeping the distraction from temperature minimal. I think including this is up to the overall design/scope of the sample set, and is ideally consistent across languages.

Reply:
@gericdong

Reply:
I just don't want anyone to get the idea that 0.4 is magic in some way. I'm okay with showing that it can be set, as long as there is a comment that prevents something thinking that the exact number is relevant.